### PR TITLE
gateway2: Fix missing liveness probe on gateway deployments

### PR DESCRIPTION
**Pull request by davidjumani**
# Description

Fixes the missing liveliness probe for the kube gateways that come up

Before : 
```
Name:             gloo-proxy-gw-868db69887-pccks
Namespace:        gloo-system
Priority:         0
Service Account:  gloo-proxy-gw
Node:             gloo-control-plane/172.18.0.3
Start Time:       Tue, 12 Nov 2024 11:55:23 -0500
Labels:           app.kubernetes.io/instance=gw
                  app.kubernetes.io/name=gloo-proxy-gw
                  gateway.networking.k8s.io/gateway-name=gw
                  gloo=kube-gateway
                  pod-template-hash=868db69887
Annotations:      prometheus.io/path: /metrics
                  prometheus.io/port: 9091
                  prometheus.io/scrape: true
Status:           Running
IP:               10.244.0.9
IPs:
  IP:           10.244.0.9
Controlled By:  ReplicaSet/gloo-proxy-gw-868db69887
Containers:
  gloo-gateway:
    Container ID:  containerd://0ed2200bd71eeef53a88b1e36c61d540a1a08cdb299e43d7fcdb5eea47dd951a
    Image:         quay.io/solo-io/gloo-envoy-wrapper:1.0.0-ci1
    Image ID:      docker.io/library/import-2024-11-12@sha256:c8f0a2be7329f67a10b08d1b1c61f10b0f8659a3c9a86b77c032e1937428959e
    Ports:         8080/TCP, 9091/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      --disable-hot-restart
      --service-node
      $(POD_NAME).$(POD_NAMESPACE)
    State:          Running
      Started:      Tue, 12 Nov 2024 11:55:24 -0500
    Ready:          True
    Restart Count:  0
    Environment:
      POD_NAME:       gloo-proxy-gw-868db69887-pccks (v1:metadata.name)
      POD_NAMESPACE:  gloo-system (v1:metadata.namespace)
      ENVOY_UID:      0
    Mounts:
      /etc/envoy from envoy-config (rw)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  envoy-config:
    Type:        ConfigMap (a volume populated by a ConfigMap)
    Name:        gloo-proxy-gw
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:          <none>

```

After :
```
Name:             gloo-proxy-gw-58c8f6d687-rg6jk
Namespace:        default
Priority:         0
Service Account:  gloo-proxy-gw
Node:             gloo-control-plane/172.18.0.3
Start Time:       Tue, 12 Nov 2024 16:44:50 -0500
Labels:           app.kubernetes.io/instance=gw
                  app.kubernetes.io/name=gloo-proxy-gw
                  gateway.networking.k8s.io/gateway-name=gw
                  gloo=kube-gateway
                  pod-template-hash=58c8f6d687
Annotations:      prometheus.io/path: /metrics
                  prometheus.io/port: 9091
                  prometheus.io/scrape: true
Status:           Running
IP:               10.244.0.10
IPs:
  IP:           10.244.0.10
Controlled By:  ReplicaSet/gloo-proxy-gw-58c8f6d687
Containers:
  gloo-gateway:
    Container ID:  containerd://bec13b8b0444b5c06736c1f0f028836cda236879cdac42ee6ecda20f653576ea
    Image:         quay.io/solo-io/gloo-envoy-wrapper:1.0.0-ci1
    Image ID:      docker.io/library/import-2024-11-12@sha256:c8f0a2be7329f67a10b08d1b1c61f10b0f8659a3c9a86b77c032e1937428959e
    Ports:         8080/TCP, 9091/TCP
    Host Ports:    0/TCP, 0/TCP
    Args:
      --disable-hot-restart
      --service-node
      $(POD_NAME).$(POD_NAMESPACE)
    State:          Running
      Started:      Tue, 12 Nov 2024 16:44:51 -0500
    Ready:          True
    Restart Count:  0
    Liveness:       exec [wget -O /dev/null 127.0.0.1:19000/ready] delay=0s timeout=1s period=10s #success=1 #failure=3.   <--------- Liveness probe
    Environment:
      POD_NAME:       gloo-proxy-gw-58c8f6d687-rg6jk (v1:metadata.name)
      POD_NAMESPACE:  default (v1:metadata.namespace)
      ENVOY_UID:      0
    Mounts:
      /etc/envoy from envoy-config (rw)
Conditions:
  Type              Status
  Initialized       True 
  Ready             True 
  ContainersReady   True 
  PodScheduled      True 
Volumes:
  envoy-config:
    Type:        ConfigMap (a volume populated by a ConfigMap)
    Name:        gloo-proxy-gw
    Optional:    false
QoS Class:       BestEffort
Node-Selectors:  <none>
Tolerations:     node.kubernetes.io/not-ready:NoExecute op=Exists for 300s
                 node.kubernetes.io/unreachable:NoExecute op=Exists for 300s
Events:
  Type    Reason     Age   From               Message
  ----    ------     ----  ----               -------
  Normal  Scheduled  16s   default-scheduler  Successfully assigned default/gloo-proxy-gw-58c8f6d687-rg6jk to gloo-control-plane
  Normal  Pulled     15s   kubelet            Container image "quay.io/solo-io/gloo-envoy-wrapper:1.0.0-ci1" already present on machine
  Normal  Created    15s   kubelet            Created container gloo-gateway
  Normal  Started    15s   kubelet            Started container gloo-gateway

```

### CODE REVIEW

**Review by ashishb-solo — COMMENTED** [review]
happy to approve as is, but any chance we can create a test for this?

File: projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml [hunk 1]
```diff
@@ -89,15 +89,16 @@ spec:
...
+        livenessProbe:
+          exec:
+            command:
+              - wget
+              - -O
+              - /dev/null
+              - 127.0.0.1:19000/ready
+            initialDelaySeconds: 3
+            periodSeconds: 10
+            failureThreshold: 3
```
Comment (nfuden) [inline projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml:101]:
```suggestion
       {- $gateway.probes.livenessprobe }
```

or something like that

Comment (davidjumani) [reply projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml:101]: Will add that once we add support for custom livenss probes

Comment (nfuden) [reply projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml:101]: but i dont think we want a default liveness at all

Comment (arianaw66) [reply projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml:101]: The issue description says "I'd like to have proper probes by default". Seems like you and Jesus (issue author) should align on desired behavior here.

Comment (jenshu) [reply projects/gateway2/helm/gloo-gateway/templates/gateway/proxy-deployment.yaml:101]: we should consider how it's done in the [edge proxies](https://github.com/solo-io/gloo/blob/main/install/helm/gloo/templates/7-gateway-proxy-deployment.yaml) and decide if we want similar configurability here

**Comment by davidjumani** [comment]
Closing in favour of https://github.com/solo-io/gloo/pull/10332

**Pull request closed without merging**